rustls/src/client/ech.rs

```diff
@@ -493,4 +561,42 @@ impl EchState {
             .collect::<Result<_, _>>()?;
         Ok(())
     }
+
+    fn server_hello_conf(server_hello: &ServerHelloPayload) -> Message {
```

Comment:
Maybe take an owned `ServerHelloPayload` if that's actually what we need, on the off chance that the caller doesn't need to keep their copy around?

Reply:
This ends up being tricky. We get our `&ServerHelloPayload` from `ExpectServerHello::handle` where presently we use `require_handshake_msg!` to extract the `&ServerHelloPayload` from `m: Message<'m>`. We read a `SupportedCipherSuite` from that `&ServerHelloPayload` and then use `m` again to start the outer hello handshake transcript with the `suite`.

If we switch to using `require_handshake_msg_move!` to get a `ServerHelloPayload` then `m` is partially moved and we can't use it to update the transcript anymore. We can't start the transcript before the move because we don't know the suite until we have a `ServerHelloPayload` to look at.

We also want the `&Message` for the new argument to `handle_server_hello` for updating the inner transcript after ECH has been confirmed, and it's a partially moved value if we pull out an owned `ServerHelloPayload` earlier.

It might be workable, but I'm pretty convinced that it will take a decent sized refactor I'm not super keen on starting at this stage. It feels like the callers always want to keep the `Message` around, payload intact.

Reply:
Ahhh, no, wasn't aware of all that complexity hanging off of that. Definitely sounds like that isn't worth it.